actors/miner/tests/move_partitions_test.rs

```diff
@@ -627,3 +628,65 @@ fn fault_and_expire_after_move() {
 
     h.check_state(&rt);
 }
+
+#[test]
+fn dispute_after_move() {
+    let (mut h, rt) = setup();
+
+    let sectors_info = h.commit_and_prove_sectors(
+        &rt,
+        1,
+        DEFAULT_SECTOR_EXPIRATION,
+        vec![vec![10], vec![20]],
+        true,
+    );
+    h.advance_and_submit_posts(&rt, &sectors_info);
+
+    let st = h.get_state(&rt);
+    let (orig_deadline_id, partition_id) =
+        st.find_sector(&rt.store, sectors_info[0].sector_number).unwrap();
+
+    h.advance_to_epoch_with_cron(&rt, nearest_safe_epoch(&rt, &h, orig_deadline_id));
+    let dest_deadline_id =
+        farthest_possible_to_deadline(&rt, orig_deadline_id, h.current_deadline(&rt));
+
+    let result = h.move_partitions(
+        &rt,
+        orig_deadline_id,
+        dest_deadline_id,
+        bitfield_from_slice(&[partition_id]),
+        || {},
+    );
+    assert!(result.is_ok());
+
+    let st = h.get_state(&rt);
+    let (dl_idx, _) = st.find_sector(&rt.store, sectors_info[0].sector_number).unwrap();
+    assert!(dl_idx == dest_deadline_id);
+
+    h.check_state(&rt);
+
+    // dispute starts here
+
+    h.advance_and_submit_posts(&rt, &sectors_info);
```

Comment:
The code below here tests that the moved sector can be disputed in its new deadline, after having been WindowPoSTed there once.

Before this, please also test that a dispute in the _old_ deadline doesn't cause any problems. You only need to model the dispute being rejected, not succeeding.

Reply:
Added a case to dispute the original empty deadline.